### PR TITLE
Support class type, key contains and key prefix queries, fixes issue #8

### PR DESCRIPTION
**Pull request by madhu314**

### CODE REVIEW

**Comment by imran0101** [comment]
@madhu314 I have added some changes to your previous commit. Please review it.

File: sample/src/main/assets/minecraft_crash.txt [hunk 1]
```diff
@@ -0,0 +1,872 @@
...
+		* inpure|api (1.7) from TWE_LIB_INpureCore-[1.7.10]1.0.0B9-65.jar
+		* LookingGlass|API (1.0) from 6_0_lookingglass-1.7.10-0.2.0.00.jar
+		* MatterOverdrive|API (0.4.1) from 2_MatterOverdrive-1.7.10-0.4.1-build-41.jar
+		* McJtyLib (1.8.1) from 2_5_1_LIB_mcjtylib-1.8.1.jar
+		* MekanismAPI|core (8.0.0) from 2_5_Networks Manager-8.2B.jar
+		* MekanismAPI|energy (9.0.0) from 2_3_MekanismAll-1.7.10-9.0.2.homebaked.jar
+		* MekanismAPI|gas (8.0.0) from 2_5_Networks Manager-8.2B.jar
+		* MekanismAPI|infuse (8.0.0) from 2_5_Networks Manager-8.2B.jar
+		* MekanismAPI|laser (8.0.0) from 2_5_Networks Manager-8.2B.jar
+		* MekanismAPI|reactor (9.0.0) from 2_3_MekanismAll-1.7.10-9.0.2.homeb
```
Comment (madhu314) [inline sample/src/main/assets/minecraft_crash.txt]: what are we using this file for ?

Comment (imran0101) [inline sample/src/main/assets/minecraft_crash.txt]: I added a large text file to add it to the db. I'll ignore this.

**Comment by madhu314** [comment]
@mipreamble Changes are good, I have altered test cases a bit...merging in changes

**Pull request merged**